shared/src/main/scala/ast/MamlKind.scala

```diff
@@ -14,6 +14,8 @@ object MamlKind {
   case object Double extends MamlKind { def repr: String = "double" }
   case object Tile extends MamlKind { def repr: String = "raster" }
   case object Geom extends MamlKind { def repr: String = "geom" }
+  // The bottom type [useful for writing inductive proofs/handling certain classes of failure]
+  case object Nothing extends MamlKind { def repr: String = "nothing" }
```

Comment:
Does this have an associated value that can be used by the user for foot-gun?

Reply:
Just trying to figure out if this is `Bottom` (inhabited) or `Void` (uninhabited).

Reply:
Good question - the main reason I'm adding it right now (and I'm sure there are implications I haven't considered) is so that I can introduce inductive proofs (for coproducts) of various behaviors that will need to return a `MamlKind` which can be reasoned about.

Reply:
Uninhabited, we decided

Reply:
Category theory thanks you.